src/Octoshift/Services/MultipartUploaderService.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web;
+using OctoshiftCLI.Extensions;
+
+namespace OctoshiftCLI.Services;
+
+public class MultipartUploaderService
+{
+    private readonly GithubClient _client;
+    internal int _streamSizeLimit = 100 * 1024 * 1024; // 100 MiB
+    internal string _base_url = "https://uploads.github.com/organizations";
+
+    public MultipartUploaderService(GithubClient client)
+    {
+        _client = client;
+    }
+
+    public async Task<string> UploadMultipart(Stream archiveContent, string archiveName, string uploadUrl)
+    {
+        var buffer = new byte[_streamSizeLimit];
+
+        try
+        {
+            // 1. Start the upload
+            var startHeaders = await StartUpload(uploadUrl, archiveName, archiveContent.Length);
```

Comment:
I would add logs to indicate progress, like for example:
`Initiating upload with Upload URL: ... and Archive Name: ...`

Reply:
I've added the parts, Im not really sure what the URL or archive name will give people